trustfall_core/src/frontend/error.rs

```diff
@@ -52,6 +52,9 @@ pub enum FrontendError {
     #[error("Incompatible types encountered in @filter: {0}")]
     FilterTypeError(#[from] FilterTypeError),
 
+    #[error("Found {0} applied to \"{1}\" property, this directive can only be applied to edges.")]
```

Comment:
Minor stylistic suggestion:
```suggestion
    #[error("Found {0} applied to \"{1}\" property, which is not supported since that directive can only be applied to edges.")]
```